pkg/controller/networkpolicy/reconciler.go

```diff
@@ -490,10 +490,16 @@ func (r *Reconciler) reconcileNetworkPolicyAllowToDNS(ctx context.Context, log l
 					// required for node local dns feature, allows egress traffic to node local dns cache
 					{
 						IPBlock: &networkingv1.IPBlock{
-							// node local dns feature is only supported for shoots with IPv4 single-stack networking
+							// node local dns feature is only supported for shoots with IPv4 or IPv6 single-stack networking
 							CIDR: fmt.Sprintf("%s/32", nodelocaldnsconstants.IPVSAddress),
 						},
 					},
+					{
+						IPBlock: &networkingv1.IPBlock{
+							// node local dns feature is only supported for shoots with IPv4 or IPv6 single-stack networking
+							CIDR: fmt.Sprintf("%s/128", nodelocaldnsconstants.IPVSIPv6Address),
+						},
+					},
```

Comment:
Do we always want to have two separate blocks or only allow what is really required? I suppose we already always enabled the link-local IPv4 address for node-local-dns regardless if it was actually needed. Hence, it might be ok to continue this approach.
WDYT?